templates/macros/list_posts.html

```diff
@@ -7,90 +7,115 @@
-                    {%- set post_index = loop.index -%}
-                    {%- set nb_posts = posts | length -%}
-                    {# in case we have a pager, the index has been computed for the current page. #}
-                    {%- if paginator -%}
-                        {%- set nb_posts = paginator.total_pages -%}
-                        {%- set number_of_other_pages = paginator.current_index - 1 -%}
-                        {%- set posts_per_page = paginator.paginate_by -%}
-                        {%- set posts_in_other_pages = number_of_other_pages * posts_per_page -%}
-                        {%- set post_index = posts_in_other_pages + post_index -%}
-                    {%- endif -%}
-                    {%- if macros_settings::evaluate_setting_priority(setting="post_listing_index_reversed", page=section, default_global_value=false) == "true" -%}
-                        {# index starts at 1 instead of 0 #}
-                        {%- set post_index = nb_posts + 1 - post_index -%}
-                    {%- endif -%}
-                    <li class="index-label">{{ post_index }}</li>
-                {%- endif -%}
-                {%- if "dates" in metadata -%}
-                    {%- set allowed_post_listing_dates = ["date", "updated", "both"] -%}
-                    {%- set post_listing_date = config.extra.post_listing_date | default(value="date") -%}
-                    {%- if post_listing_date not in allowed_post_listing_dates -%}
-                        {{ throw(message="ERROR: Invalid value for config.extra.post_listing_date. Allowed values are 'date', 'updated', or 'both'.") }}
+            <section class="bloglist-meta {% if bottom_divider -%}bottom-divider{%- endif -%}">
+                <ul>
+                    {%- if "indexes" in metadata -%}
+                        {%- set post_index = loop.index -%}
+                        {%- set number_of_posts = posts | length -%}
+                        {# in case we have a pager, the index has been computed for the current page. #}
+                        {%- if paginator -%}
+                            {%- set number_of_posts = paginator.total_pages -%}
+                            {%- set number_of_other_pages = paginator.current_index - 1 -%}
+                            {%- set posts_per_page = paginator.paginate_by -%}
+                            {%- set posts_in_other_pages = number_of_other_pages * posts_per_page -%}
+                            {%- set post_index = posts_in_other_pages + post_index -%}
+                        {%- endif -%}
+                        {%- if macros_settings::evaluate_setting_priority(setting="post_listing_index_reversed", page=section, default_global_value=false) == "true" -%}
+                            {# index starts at 1 instead of 0 #}
+                            {%- set post_index = number_of_posts + 1 - post_index -%}
+                        {%- endif -%}
+                        <li>{{ post_index }}</li>
```

Comment:
The `class="index-label"` was there for users to easily style the index label if needed (even if by default there is no specific styling needed).
It was not documented like the other CSS classes as I though users will just look at the dom and create a custom CSS file accordingly.
As index, date and draft labels are all `<li>` of the same `<ul>` and can be optional, I am not sure how easy it could be to style the index label without impacting the others (in particular when the index one is not there).

Anyway, I let you decide if this reason is enough to keep this CSS class.
If yes, adding a comment to explain why we have an orphaned CSS class would be worth it here.

Reply:
Makes sense! I've added it back. Thanks.